### PR TITLE
Localization context for linguists

### PR DESCRIPTION
**Pull request by tylerrowsell**
Closes https://github.com/Shopify/custom-data-issues/issues/2940

- Adds a script to generate english distributions from `/data` and add context for linguists.
- Updates make files to include sentinel for changes
- Updates english using above script.
- Updates all other localization files with comment that they're automatically generated via github action. (we're not there yet but will be soon).

### CODE REVIEW

File: data/localizations/attributes/de.yml [hunk 1]
```diff
@@ -1,3 +1,4 @@
+# This file is auto-generated through a github action. Do not edit directly.
```
Comment (KComrade53) [inline data/localizations/attributes/de.yml:1]: Does this mean public contributors will no longer be able to submit translations suggestions via PR? Would they just do so via an issue then if they have translation suggestions?

Comment (tylerrowsell) [reply data/localizations/attributes/de.yml:1]: Yes that's correct. We'll want to confirm any localization with the linguists so we'd prefer to go through an issue.

File: bin/sync_en_localizations [hunk 2]
```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require "yaml"
+require "fileutils"
+require_relative "../config/environment"
```
Comment (elsom25) [inline bin/sync_en_localizations:6]: We have a CLI helper — let's use that in this file as well?

**Review by elsom25 — APPROVED** [review]
Good to go, though would like to have the bin file use our CLI helper before we call it _done_

**Pull request merged**